compiler/rustc_passes/src/stability.rs

```diff
@@ -140,6 +156,8 @@ impl<'a, 'tcx> Annotator<'a, 'tcx> {
             const_stab
         });
 
+        // `impl const Trait for Type` items forward their const stability to their
+        // immediate children.
         if const_stab.is_none() {
```

Comment:
So we still do all the work of recursing but then throw away the result most of the time? Isn't that rather wasteful?

Reply:
not really, all that happens is that at the recursion site we store an `Option<&'tcx ConstStability>` in the visitor datastructure, which almost always is `None`. I do not think this has any effect on performance as the entire stability checking is very cheap

Reply:
Fair.

I don't actually understand any of the code here.^^